### PR TITLE
CC-1029: Use rotating badge for earned badge banner.

### PR DESCRIPTION
**Pull request by ryan-gang**
**Checklist**:

- [X] I've thoroughly self-reviewed my changes
- [X] I've added tests for my changes, unless they affect admin-only areas (or are otherwise not worth testing)
- [ ] I've verified any visual changes using Percy (add a commit with `[percy]` in the message to trigger)

## Summary by CodeRabbit

- **New Features**
	- Updated the earned badge notice with improved styling and functionality. Now includes a link to read the story behind the badge.
- **Style**
	- Added new animation effects (`wiggle` and `spin-slow`) for enhanced visual appeal.
- **Tests**
	- Modified acceptance tests to reflect changes in badge notice text and interaction methods.
- **Refactor**
	- Simplified interaction with earned badge notices by removing unnecessary button and streamlining click handling.

### CODE REVIEW

File: tests/pages/course-page.js [hunk 1]
```diff
@@ -105,7 +105,7 @@ export default create({
       scope: '[data-test-badge-earned-modal]',
     },
 
-    clickOnViewButton: clickable('[data-test-view-button]'),
+    clickOnViewButton: clickable(),
```
Comment (rohitpaulk) [inline tests/pages/course-page.js:108]: There's no "view" button now no? Can remove this and just use `earnedBadgeNotice.click()`.

File: tailwind.config.js [hunk 2]
```diff
@@ -200,10 +200,15 @@ module.exports = {
           '50%': { opacity: '0' },
           '100%': { opacity: '1' },
         },
+        wiggle: {
```
Comment (rohitpaulk) [inline tailwind.config.js:203]: Where is this used

File: app/components/course-page/course-stage-step/earned-badge-notice.hbs [hunk 3]
```diff
@@ -1,31 +1,25 @@
 {{! @glint-nocheck: not typesafe yet }}
 <div
-  class="border rounded flex items-center justify-between p-4 border-gray-800 hover:border-[#3F3F3F] bg-gradient-to-br from-[#383838] to-black from-0% to-100%"
+  class="border rounded flex items-center justify-between border-gray-800 hover:border-[#3F3F3F] bg-gradient-to-br from-[#383838] to-black from-0% to-100% cursor-pointer"
+  {{on "click" this.handleViewButtonClicked}}
```
Comment (rohitpaulk) [inline app/components/course-page/course-stage-step/earned-badge-notice.hbs:4]:
```suggestion
  {{on "click" this.handleClick}}
```

(There's no "view button" anymore)

File: app/components/course-page/course-stage-step/earned-badge-notice.hbs [hunk 4]
```diff
@@ -1,31 +1,27 @@
 {{! @glint-nocheck: not typesafe yet }}
```
Comment (rohitpaulk) [inline app/components/course-page/course-stage-step/earned-badge-notice.hbs:1]: @ryan-gang when editing files, I'd recommend converting them to typescript as you go - ensures that we aren't introducing new bugs that we could've caught using types. 

(Only exception here would be if the file is huge and the change is minimal)

**Review by rohitpaulk — APPROVED** [review]

File: tailwind.config.js [hunk 5]
```diff
@@ -200,10 +200,15 @@ module.exports = {
...
+          '0%, 100%': { transform: 'rotate(-25deg)' },
+          '50%': { transform: 'rotate(25deg)' },
+        },
       },
       animation: {
         'infinite-slide': 'infinite-slide 30s linear infinite',
         'blink-fast': 'blink 1s step-start infinite',
+        'spin-slow': 'spin 12s linear infinite',
```
Comment (rohitpaulk) [inline tailwind.config.js:211]: @ryan-gang I think this was supposed to be 16s - can you confirm with Louis once please?

File: tailwind.config.js (hunk 2 above)
Comment (ryan-gang) [reply tailwind.config.js:203]: Was used in the previous animation, removed this block now.

**Pull request merged**